src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/personalrelationships/profiledetails/ContactPersonProfileDetailsSyncService.kt

```diff
@@ -16,41 +24,62 @@ class ContactPersonProfileDetailsSyncService(
-        ),
-        null,
-      )
+      ignoreTelemetry(prisonerNumber, domesticStatusId, MARITAL)
+    }
+  }
+
+  suspend fun syncNumberOfChildren(event: ContactNumberOfChildrenCreatedEvent) {
+    val prisonerNumber = event.personReference.identifiers.first { it.type == "prisonerNumber" }.value
+    val numberOfChildrenId = event.additionalInformation.numberOfChildrenId
+    if (event.additionalInformation.source == "DPS") {
+      syncProfileDetail(prisonerNumber, numberOfChildrenId, CHILD)
+    } else {
+      ignoreTelemetry(prisonerNumber, numberOfChildrenId, CHILD)
     }
   }
 
-  suspend fun syncDomesticStatus(prisonerNumber: String, domesticStatusId: Long) {
+  private fun ignoreTelemetry(prisonerNumber: String, dpsId: Long, profileType: ContactPersonProfileType) = telemetryClient.trackEvent(
+    "contact-person-${profileType.identifier}-ignored",
+    mapOf(
+      "offenderNo" to prisonerNumber,
+      "dpsId" to dpsId.toString(),
+      "reason" to "Entity was created in NOMIS",
+    ),
+    null,
+  )
+
+  private suspend fun syncProfileDetail(prisonerNumber: String, dpsId: Long, profileType: ContactPersonProfileType) {
     val telemetry = mutableMapOf(
       "offenderNo" to prisonerNumber,
-      "domesticStatusId" to domesticStatusId.toString(),
+      "dpsId" to dpsId.toString(),
     )
     runCatching {
-      dpsApi.getDomesticStatus(prisonerNumber)
-        .also { telemetry["dpsId"] = it.id.toString() }
-        .let { nomisApi.upsertProfileDetails(prisonerNumber, "MARITAL", it.domesticStatusCode) }
-        .also { telemetry["bookingId"] = it.bookingId.toString() }
-        .also {
-          telemetryClient.trackEvent(
-            """contact-person-domestic-status-${if (it.created) "created" else "updated"}""",
-            telemetry,
-            null,
-          )
-        }
+      val (bookingId, created) = performSync(profileType, prisonerNumber)
+      telemetry["bookingId"] = bookingId.toString()
+      telemetryClient.trackEvent(
+        """contact-person-${profileType.identifier}-${if (created) "created" else "updated"}""",
+        telemetry,
+        null,
+      )
     }.onFailure { e ->
       telemetry["error"] = e.message ?: "unknown"
-      telemetryClient.trackEvent("contact-person-domestic-status-error", telemetry, null)
+      telemetryClient.trackEvent("contact-person-${profileType.identifier}-error", telemetry, null)
       throw e
     }
   }
+
+  private suspend fun performSync(profileType: ContactPersonProfileType, prisonerNumber: String) = when (profileType) {
+    MARITAL -> {
+      dpsApi.getDomesticStatus(prisonerNumber)
+        .let { nomisApi.upsertProfileDetails(prisonerNumber, "MARITAL", it.domesticStatusCode) }
+        .let { it.bookingId to it.created }
+    }
+    CHILD -> {
+      dpsApi.getNumberOfChildren(prisonerNumber)
+        .let { nomisApi.upsertProfileDetails(prisonerNumber, "CHILD", it.numberOfChildren) }
+        .let { it.bookingId to it.created }
+    }
```

Comment:
Changed the service to work for a generic profile detail sync, with the different API calls captured here